### PR TITLE
[MIG] migration pattern

### PR DESCRIPTION
**Pull request by sebastienbeau**

### CODE REVIEW

**Comment by sebastienbeau** [comment]
ocabot merge nobump

File: pattern_import_export/models/ir_exports_line.py [hunk 1]
```diff
@@ -142,9 +142,9 @@ def _check_required_fields(self):
     @api.depends("name")
     def _compute_related_level_field(self):
         for export_line in self:
-            if export_line.pattern_config_id.model_id.model and export_line.name:
+            if export_line.name:
```
Comment (kevinkhao) [inline pattern_import_export/models/ir_exports_line.py:145]: additional condition of export_line.pattern_config_id.model_id.model is needed. Without it we get a crash.

File: pattern_import_export/models/ir_exports_line.py [hunk 2]
```diff
@@ -142,9 +142,9 @@ def _check_required_fields(self):
     @api.depends("name")
     def _compute_related_level_field(self):
         for export_line in self:
-            if export_line.pattern_config_id.model_id.model and export_line.name:
```
Comment (kevinkhao) [inline pattern_import_export/models/ir_exports_line.py:144]: This is reponsible for the problem with creating new patterns and crash at L149:
- at L146 we don't get the expected values (model is false)
- this is due to the fact that our new() record that doesn't have the values we expect, yet

This _compute_related_level_field compute function is responsible for setting the values for the fields:

related_model_id
last_field_id
level

These fields are not visible to the user and are used only during export (or import), so they are not relevant during the time the user is using the UI to create one. In other words they could be used in a write() function.

**Pull request merged**